assets/src/blocks/Counter/CounterBlock.js

```diff
@@ -0,0 +1,106 @@
+import {CounterEditor} from './CounterEditor';
+import {frontendRendered} from '../frontendRendered';
+import {CounterFrontend} from './CounterFrontend';
+import {renderToString} from 'react-dom/server';
+
+export const BLOCK_NAME = 'planet4-blocks/counter';
+
+const attributes = {
+  title: {
+    type: 'string',
+    default: '',
+  },
+  description: {
+    type: 'string',
+    default: '',
+  },
+  completed: {
+    type: 'integer',
+    default: '',
+  },
+  completed_api: {
+    type: 'string',
+    default: '',
+  },
+  target: {
+    type: 'integer',
+    default: '',
+  },
+  text: {
+    type: 'string',
+    default: '',
+  },
+  style: { // Needed to convert existing blocks
+    type: 'string',
+    default: '',
+  },
+};
+
+export const registerCounterBlock = () => {
+  const {registerBlockType, unregisterBlockStyle, registerBlockStyle} = wp.blocks;
+  const {RawHTML} = wp.element;
+
+  registerBlockType(BLOCK_NAME, {
+    title: 'Counter',
+    icon: 'dashboard',
+    category: 'planet4-blocks',
+    attributes,
+    supports: {
+      html: false, // Disable "Edit as HTMl" block option.
+    },
+    // eslint-disable-next-line no-shadow
+    edit: CounterEditor,
+    save: props => {
+      const markup = renderToString(
+        <div
+          data-hydrate={BLOCK_NAME}
+          data-attributes={JSON.stringify(props.attributes)}
+        >
+          <CounterFrontend {...props.attributes} />
+        </div>
+      );
+      return <RawHTML>{markup}</RawHTML>;
+    },
+    deprecated: [
+      {
+        attributes,
+        save: frontendRendered(BLOCK_NAME),
+      },
+      {
+        attributes,
+        save() {
+          return null;
+        },
+      },
+    ],
+  });
+
+  // Remove the default style since it's the same as "text only"
+  unregisterBlockStyle(BLOCK_NAME, 'default');
+
+  const styles = [
+    {
+      name: 'plain',
+      label: 'Text Only',
+      isDefault: true,
+    },
+    {
+      name: 'bar',
+      label: 'Progress Bar',
+    },
+    {
+      name: 'arc',
+      label: 'Progress Dial',
+    },
+
+  ];
+
+  if (window.p4ge_vars.features.feature_engaging_networks) {
```

Comment:
These variables are set in the blocks repo, if we need them here we should set them in the theme instead? I've had to do [something similar](https://github.com/greenpeace/planet4-master-theme/pull/2167/commits/0e645a1e81fbed6fa407c5ca79aafbb8332fd0e4) for the Cookies block but I'm not 100% sure it's the right approach, waiting for feedback

Reply:
I pushed [this proposal](https://github.com/greenpeace/planet4-master-theme/pull/2176/commits/dfca81dfefd998db6898264ce35d84a35b07e560) and includes features too.

If you agree, then we might need to update the Cookies block ([here is the commit](https://github.com/greenpeace/planet4-master-theme/pull/2176/commits/82b5c1866a4801cdfa0d27b76bed8fef99329619)). In that case, I'd prefer to merge it separately.